### PR TITLE
Collect codecoverage info from jobs for all database types.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
Hmm.. The PHP 7 jobs for codecoverage are getting killed. Checking travis docs most likely cause is running out of memory.

**Comment by thinkingmedia** [comment]
Travis containers have 3GB which should be enough. You might need to update the PHP.ini to increase memory.

You can try this.

```
echo "memory_limit=1024M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
```

Also, the new Trusty image has 7GB of memory.

https://docs.travis-ci.com/user/ci-environment/

**Comment by ADmad** [comment]
Increasing memory to 2GB doesn't help, so I don't think it's memory related.

**Comment by thinkingmedia** [comment]
Run it with `phpunit --verbose --debug` so we can see which test is the last one. Looks like it dies at the same point each time.

**Comment by ADmad** [comment]
@thinkingmedia Seems to always die at `Cake\Test\TestCase\Shell\Task\ExtractTaskTest::testExtractCore`.  I am taking a break from trying to please the machines. You can take the diff of this PR and tinker around with it if you like :)

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 language: php
 
 php:
-  - 5.5
-  - 5.6
```
Comment (markstory) [inline .travis.yml]: Not sure this is a good idea.

Comment (ADmad) [inline .travis.yml]: @markstory That's just temporary so that I could php 7 jobs running quickly as they are the ones having problem.

Comment (markstory) [inline .travis.yml]: ok.

**Comment by ADmad** [comment]
This is good to go. I'll further try and tweak codecov settings in a separate PR.

Btw I like this codecoverage info comment. Codecov smartly only makes 1 comment per PR and updates the same when updates are done to the PR, thereby not flooding us with notifications.

File: .travis.yml [hunk 2]
```diff
@@ -82,13 +78,13 @@ before_script:
   - composer install --prefer-dist --no-interaction
 
 script:
-  - sh -c "if [ '$DEFAULT' = '1' ]; then vendor/bin/phpunit; fi"
+  - if [[ $DEFAULT = 1 && $TRAVIS_PHP_VERSION = 7.* ]]; then export CODECOVERAGE=1 ; phpdbg -qrr vendor/bin/phpunit --coverage-clover=clover.xml; fi
```
Comment (markstory) [inline .travis.yml]: Nice. Saves a build in the matrix.

**Comment by lorenzo** [comment]
with this change + the one running the builds with `sudo: required` we are back to the bad spot in which we need to wait around 30 minutes to see a build pass :(

**Comment by ADmad** [comment]
One cause of slowness is the "trusty" VMs don't have PHP pre-installed so for each job they are installed through phpenv. Travis [docs](https://docs.travis-ci.com/user/trusty-ci-environment#PHP) says having them pre-installed is at top of their list for next update so hopefully it should be available soon.

I'll see if there are other tweaks that can be done to speed up things.

**Comment by thinkingmedia** [comment]
They did publish that update but there were bugs in it being reports so Travis pulled back the update. I don't know why they will have it fixed but it should be soon.

**Pull request merged**